### ISSUE TITLE
Driver will download again even exists in local

### ISSUE THREAD

**Issue by xiaohuihui093** [SergeyPirogov/webdriver_manager#612]
In core/driver_cache.py line 108, self.__get_metadata_key(driver) will return None because no return value in self.__get_metadata_key method.

**Comment by Tech-Goddess** [comment]
We encountered the same problem in both MacOS and Windows. It seems that the chrome webdriver zip now has a folder within the archive, so the check if the version exists always returns false.

Up until chrome version 114, the driver binary used to get extracted to a path like:
`C:\Users\user\.wdm\drivers\chromedriver\win32\108.0.5359\chromedriver.exe
`

starting from 115, it's now:
`C:\Users\user\.wdm\drivers\chromedriver\win64\116.0.5845.112\chromedriver-win32\chromedriver.exe
`

the driver seems to work just fine after download, but every time we start a new instance, the driver isn't located in local.

Also we've noticed the newest paths returned from DriverCacheManager.load_metadata_content() aren't concatenated correctly, see binary_path in.

`'win64_chromedriver_116.0.5845.112_for_116.0.5845.112' = {'timestamp': '31/08/2023', 'binary_path': 'C:\\Users\\user\\.wdm\\drivers\\chromedriver\\win64\\116.0.5845.112\\chromedriver-win32/chromedriver.exe'}`

**Comment by burrk** [comment]
For the problem @xiaohuihui093 reported __get_metadata_key(driver) only fails the first time it is called.  We worked around it by cheating a bit (until a fix is available)
```python
wdm = ChromeDriverManager(driver_version=self.driver_version)
# Pre-initialize the driver's cache manager
wdm._cache_manager.find_driver(wdm.driver)
self.driver_path = wdm.install()
```

@Tech-Goddess, you may have a different problem?

**Comment by SergeyPirogov** [comment]
Fixed in 4.0.1
